qiskit/primitives/statevector_sampler.py

```diff
@@ -52,6 +52,73 @@ class _MeasureInfo:
 class StatevectorSampler(BaseSamplerV2):
     """
     Simple implementation of :class:`BaseSamplerV2` using full state vector simulation.
+
+    This class is implemented via :class:`~.Statevector` which turns provided circuits into
+    pure state vectors, and is therefore incompatible with mid-circuit measurements (although
+    other implementions may be).
+
+    As seen in the example below, this sampler supports providing arrays of parameter value sets to
+    bind against a single circuit.
+
+    Each tuple of ``(circuit, <optional> parameter values, <optional> shots)``, called a sampler
+    primitive unified bloc (PUB), produces its own array-valued result. The :meth:`~run` method can
+    be given many pubs at once.
+
+    .. code-block:: python
+
+        from qiskit.circuit import Parameter, QuantumCircuit, ClassicalRegister, QuantumRegister
+        from qiskit.primitives import StatevectorSampler
+
+        import matplotlib.pyplot as plt
+        import numpy as np
+
+        # Define our circuit registers, including classical registers called 'alpha' and 'beta'.
+        qreg = QuantumRegister(3)
+        alpha = ClassicalRegister(2, "alpha")
+        beta = ClassicalRegister(1, "beta")
+
+        # Define a quantum circuit with two parameters.
+        circuit = QuantumCircuit(qreg, alpha, beta)
+        circuit.h(0)
+        circuit.cx(0, 1)
+        circuit.cx(1, 2)
+        circuit.ry(Parameter("a"), 0)
+        circuit.rz(Parameter("b"), 0)
+        circuit.cx(1, 2)
+        circuit.cx(0, 1)
+        circuit.h(0)
+        circuit.measure([0, 1], alpha)
+        circuit.measure([2], beta)
+
+        # Define a sweep over parameter values, where the second axis is over
+        # the two parameters in the circuit.
+        params = np.vstack([
+            np.linspace(-np.pi, np.pi, 100),
+            np.linspace(-4 * np.pi, 4 * np.pi, 100)
+        ]).T
+
+        # Instantiate a new statevector simulation based sampler object.
+        sampler = StatevectorSampler()
+
+        # Start a job that will return shots for all 100 parameter value sets.
+        pub = (circuit, params)
+        job = sampler.run([pub], shots=256)
+
+        # Extract the result for the 0th pub (this example only has one pub).
+        result = job.result()[0]
+
+        # There is one BitArray object for each ClassicalRegister in the circuit.
+        # Here, we can see that the BitArray for alpha contains data for all 100 sweep points,
+        # and that it is indeed storing data for 2 bits over 256 shots.
+        assert result.data.alpha.shape == (100,)
+        assert result.data.alpha.num_bits == 2
+        assert result.data.alpha.num_shots == 256
+
+        # We can turn the data from the 22nd sweep index into a counts dict.
+        result.data.alpha.get_counts(22)
+
+        # Or, for low-level applications, we can work directly with the binary data.
+        result.data.alpha.array
```

Comment:
Bitstrings as str #11700 might make the result more intuitive

main branch
```
print(result.data.alpha.array)

[[[3]
  [3]
  [3]
  ...
  [3]
  [3]
  [3]]

 ...

 [[3]
  [3]
  [3]
  ...
  [3]
  [3]
  [3]]]
```

#11700
```
print(result.data.alpha.get_bitstrings())

['11', '11', '11', '11', '11', '11', ... (many bitstrings) ..., '11', '11', '11', '11']
```

Reply:
Done in [ed11f05](https://github.com/Qiskit/qiskit/pull/11589/commits/ed11f053dfc79b020a5df9d2169a487399a6fe79)